### PR TITLE
[G2M] Bar-Line chart 

### LINKED ISSUE #177: Bar Chart with Trend Line
image

### PR DESCRIPTION
**Pull request by kc-leung**
fixes #177 

- Added plotly BarLine chart

### CODE REVIEW

File: src/components/plotly/bar-line/index.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      extra: {
+        mode: 'lines',
+        fill: 'tonexty',
+      },
+      ...props,
+    })[0],
+  }
+
+  return (
+    <>
```
Comment (woozyking) [inline src/components/plotly/bar-line/index.js:48]: the fragment here isn't necessary right?

File: src/components/plotly/bar-line/index.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          },
+        }}
+      />
+    </>
+  )
+}
+
+BarLine.propTypes = {
+  x: PropTypes.string.isRequired,
+  y: PropTypes.arrayOf(PropTypes.string).isRequired,
```
Comment (woozyking) [inline src/components/plotly/bar-line/index.js:92]: can we confine this further to say it's an array of strings with either 1 or 2 elements?

**Review by woozyking — APPROVED** [review]
looks good. a few comments

File: src/components/plotly/bar-line/index.js (hunk 2 above)
Comment (kc-leung) [reply src/components/plotly/bar-line/index.js:92]: yea for sure

**Review by geoerika — APPROVED** [review]
Very cool!

**Review by vlatawiec — APPROVED** [review]
Looks good! If possible in the future, the Legend section can be centered within the Widget (better visibility + a more pleasant UI) Otherwise, should work as needed for V1, great job!

**Comment by kc-leung** [comment]
> Looks good! If possible in the future, the Legend section can be centered within the Widget (better visibility + a more pleasant UI) Otherwise, should work as needed for V1, great job!

I will try to make it in my next PR not sure if plotly allows it

**Comment by geoerika** [comment]
@kc-leung is your local story working? because in Chromatic there seems to be an error:
Screen Shot 2022-10-11 at 10 08 37 AM

**Comment by kc-leung** [comment]
> @kc-leung is your local story working? because in Chromatic there seems to be an error: Screen Shot 2022-10-11 at 10 08 37 AM

need to rebase it yea. We have made a lot of changes for COX

**Pull request merged**